### PR TITLE
 Add review_submitted handler which changes labels after a review

### LINKED ISSUE #1373: change PR to "waiting on author" after "changes requested" review
It would be useful if rustbot changed the status of a PR to `S-waiting-on-author` after observing a "changes requested" review from the assigned reviewer. 

Branched out from https://github.com/rust-lang/triagebot/issues/1339

### PR DESCRIPTION
**Pull request by mibac138**
Fixes #1373 
cc @nikomatsakis

### CODE REVIEW

**Comment by bjorn3** [comment]
I think an approving review shouldn't mark the PR as waiting on author.

**Comment by mibac138** [comment]
Didn't consider that case, thanks. I've added it now

File: src/github.rs [hunk 1]
```diff
@@ -292,6 +292,18 @@ pub struct Comment {
...
+}
+
+#[serde(rename_all = "SCREAMING_SNAKE_CASE")]
+#[derive(Debug, serde::Deserialize)]
+pub enum PullRequestReviewState {
+    Approved,
+    ChangesRequested,
+    Commented,
+    Dismissed,
+    Pending,
```
Comment (mibac138) [inline src/github.rs:306]: I *think* this is correct, but I'd appreciate if this was double checked before merging

**Comment by nikomatsakis** [comment]
Oooh, nice!

File: src/handlers/review_submitted.rs [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+            },
+            ..
+        },
+    ) = event
+    {
+        if event.comment.pr_review_state == PullRequestReviewState::Approved {
+            return Ok(());
+        }
+
+        if event.issue.assignees.contains(&event.comment.user) {
```
Comment (nikomatsakis) [inline src/handlers/review_submitted.rs:26]: I think we should only do this for "Request Changes". I'm not sure how to interpret a simple "comment" review that has no "inclination".

**Review by nikomatsakis — CHANGES_REQUESTED** [review]

File: src/handlers/review_submitted.rs (hunk 2 above)
Comment (mibac138) [reply src/handlers/review_submitted.rs:26]: Makes sense, changed it

**Review by nikomatsakis — APPROVED** [review]

**Review by nikomatsakis — APPROVED** [review]

File: src/github.rs [hunk 3]
```diff
@@ -292,6 +292,18 @@ pub struct Comment {
     pub user: User,
     #[serde(alias = "submitted_at")] // for pull request reviews
     pub updated_at: chrono::DateTime<Utc>,
+    #[serde(rename = "state")]
+    pub pr_review_state: PullRequestReviewState,
```
Comment (Mark-Simulacrum) [inline src/github.rs:296]: I believe this may have broken deserialization of issue comments -- I'm filing a fix now.

**Pull request merged**